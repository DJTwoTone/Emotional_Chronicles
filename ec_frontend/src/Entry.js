import React from 'react';

function Entry () {

    return (
        <div>
        Entry goes here.    
        </div>
            )
}

export default Entry;