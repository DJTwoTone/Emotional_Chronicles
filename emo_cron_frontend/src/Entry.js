import React, { useContext, useEffect, useState } from 'react';
import { useParams } from 'react-router';
import { DateTime } from 'luxon'

import ECApi from './ECApi';

import UserContext from './UserContext';

function Entry() {

    let { username, date } = useParams();

    const [displayedEntry, setDisplayedEntry] = useState({})

    useEffect(() => {
        async function fetchDiaryEntry() {
            try {
                let convertedDate = new DateTime(date).toSQLDate();
                let entryData = await ECApi.getEntry(username, convertedDate);
                console.log('entryDate', entryData)
                setDisplayedEntry(entryData)
            }  catch (err) {
                console.log(err)
            }
        }
        fetchDiaryEntry();
    }, [date, username])




    return (

        <div>
            {JSON.stringify(displayedEntry)}
        </div>



    )


}

export default Entry;