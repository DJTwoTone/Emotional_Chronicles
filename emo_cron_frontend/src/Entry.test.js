import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Entry from './Entry';

it('should render', function() {
    render(
        <MemoryRouter>
            <Entry />
        </MemoryRouter>
    );
})

it('should match the snapshot', function() {
    const { asFragment } = render(
        <MemoryRouter>
            <Entry />
        </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
})