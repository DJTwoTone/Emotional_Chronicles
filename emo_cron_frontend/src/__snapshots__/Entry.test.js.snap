// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="justify-content-center m-5 container"
  >
    <div
      class="card"
    >
      <div
        class="card-header"
      >
        Invalid DateTime
      </div>
      <div
        class="card-body"
      >
        <p
          class="card-text"
        >
          You wrote:
        </p>
        <p
          class="card-text"
        />
        <p
          class="card-text"
        >
          Your writing prompt was:
        </p>
        <p
          class="card-text"
        />
        <p
          class="card-text"
        >
          You may have been inspired by:
        </p>
        <p
          class="card-text"
        />
        <div
          class="row g-1 justify-content-around"
        >
          <div
            class="col-5"
          >
            <p
              class="card-text"
            >
              You said you were feeling:
            </p>
            <p
              class="card-text"
            >
              You didn't share this with us
            </p>
          </div>
          <div
            class="col-5"
          >
            <p
              class="card-text"
            >
              Our analysis shows you might be feeling:
            </p>
            <div
              class="list-group"
            />
          </div>
        </div>
        <div>
          <p>
            Emotional Radar
          </p>
          <div
            class="recharts-responsive-container"
            style="width: 75%; height: 300px;"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
